.github/actions/restore-cache/action.yml

```
@@ -11,8 +15,9 @@ runs:
 
-    - name: Check if caches are restored
-      uses: actions/github-script@v6
+    - name: Install dependencies on cache miss
       if: steps.dep-cache.outputs.cache-hit != 'true'
-      with:
-        script: core.setFailed('Dependency cache could not be restored - please re-run ALL jobs.')
\ No newline at end of file
+      shell: bash
+      run: |
+        echo "::warning::Dependency cache could not be restored - installing dependencies from scratch"
+        yarn install --prefer-offline --frozen-lockfile
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)